gcloud/datastore/connection.py

```diff
@@ -149,7 +149,7 @@ def mutation(self):
                   (if one exists) or or a new mutation instance.
         """
         if self.transaction():
-            return self.transaction().mutation()
+            return self.transaction().mutation
```

Comment:
@tseaver I didn't catch this bug (in the first commit) because of all the inline `Xact` classes with `Xact.mutation()` defined.

Do you think we need a deep dive of our tests and put all the mock classes in one place while simultaneously reducing their use? (After the API changes are all complete.)